distributed/tests/test_client.py

```diff
@@ -6791,6 +6791,18 @@ def setup(self, worker=None):
         await c.register_worker_plugin(MyPlugin())
 
 
+@gen_cluster(client=True)
+async def test_scheduler_plugins(c, s, a, b):
```

Comment:
Tests are currently spread around a bit, but most tests for the `SchedulerPlugin` can be found in `distributed/diagnostics/tests/test_scheduler_plugin.py`. I'd suggest moving this test over for better discovery.